src/Tribe/Widget/Widget_Abstract.php

```diff
@@ -0,0 +1,416 @@
+ *
+ * @since   TBD
+ * @package Tribe\Widget
+ *
+ */
+abstract class Widget_Abstract extends \WP_Widget implements Widget_Interface {
+
+	/**
+	 * Slug of the current widget.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	protected $slug;
+
+	/**
+	 * The view interface for the widget.
+	 *
+	 * @var View_Interface;
+	 */
+	protected $view;
+
+	/**
+	 * The slug of the widget view.
+	 *
+	 * @var string
+	 */
+	protected $view_slug;
+
+	/**
+	 * Default arguments to be merged into final arguments of the widget.
+	 *
+	 * @since TBD
+	 *
+	 * @var array
+	 */
+	protected $default_arguments = [];
+
+	/**
+	 * Array map allowing aliased widget arguments.
+	 *
+	 * The array keys are aliases of the array values (i.e. the "real" widget attributes to parse).
+	 * Example array: [ 'alias' => 'canonical', 'from' => 'to', 'that' => 'becomes_this' ]
+	 * Example widget usage: [some_tag alias=17 to='Fred'] will be parsed as [some_tag canonical=17 to='Fred']
+	 *
+	 * @since TBD
+	 *
+	 * @var array<string,string>
+	 */
+	protected $aliased_arguments = [];
+
+	/**
+	 * Array of callbacks for arguments validation.
+	 *
+	 * @since TBD
+	 *
+	 * @var array
+	 */
+	protected $validate_arguments_map = [];
+
+	/**
+	 * Arguments of the current widget.
+	 *
+	 * @since TBD
+	 *
+	 * @var array
+	 */
+	protected $arguments;
+
+	/**
+	 * Content of the current widget.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	protected $content;
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function __construct( $id_base = '', $name = '', $widget_options = [], $control_options = [] ) {
+
+		$arguments = $this->get_arguments();
+
+		parent::__construct(
+			Arr::get( $arguments, 'id_base', '' ),
+			Arr::get( $arguments, 'name', '' ),
+			Arr::get( $arguments, 'widget_options', [] ),
+			Arr::get( $arguments, 'control_options', [] )
+		);
+
+		// Setup a view instance for the widget.
+		$this->setup_view();
+
+		// todo add what this does in in TEC-3612 & TEC-3613
+		$this->setup();
+	}
+
+	/**
+	 * Setup the widget.
+	 *
+	 * @todo update in TEC-3612 & TEC-3613
+	 *
+	 * @since TBD
+	 *
+	 * @return mixed
+	 */
+	public abstract function setup();
+
+	/**
+	 * Setup the view for the widget.
+	 *
+	 * @since TBD
+	 */
+	public function setup_view() {
+		$context = tribe_context();
+
+		// Modifies the Context for the shortcode params.
+		$context = $this->alter_context( $context );
+
+		// Setup the view instance.
+		$view = View::make( $this->get_view_slug(), $context );
+
+		$view->get_template()->set_values( $this->get_arguments(), false );
+
+		$this->set_view( $view );
+	}
+
+	/**
+	 * Echoes the widget content.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $args     Display arguments including 'before_title', 'after_title',
+	 *                        'before_widget', and 'after_widget'.
+	 * @param array $instance The settings for the particular instance of the widget.
+	 */
+	public function widget( $args, $instance ) {
+		$html = $this->get_html();
```

Comment:
Should this just echo, or will there be a filter added later?

Reply:
Not sure yet, this area will be developed in later tickets.